doc/userguide/src/CreatingTestData/AdvancedFeatures.rst

```diff
@@ -399,6 +399,103 @@ lower limit, upper limit and step.
    \                                  Log          ${index}
    ================  ===============  ===========  =========  ===========  ========  =======
 
+For in Enumerate
+~~~~~~~~~~~~~~~~
+
+Sometimes it is useful to loop over a list and also keep track of your location
+inside the list.  Robot Framework has a special
+`FOR index ... IN ENUMERATE ...` syntax for this situation.
+This syntax is derived from the
+`Python built-in function <https://docs.python.org/2/library/functions.html#enumerate>`.
```

Comment:
I don't think this creates a link. Have you tested generating User Guide to see does it work?